### ISSUE TITLE
[Indigo] [General] Prepare to merge into `main` for sprint 6.

### ISSUE THREAD

**Issue by lilingxi01** [david-fisher/320-S22-Track2#124]
# Description

Prepare to merge current branch into `main` periodically. But before that, we need to make sure that the content in `indigo-dev` is stable enough. To do that, we are planning to merge only after our Cypress tests are covering most of the components (at least 50%).

# Checklist

- [x] ~~Integrate code coverage detection into Cypress.~~
  - It is not really convenient to achieve and I do not want to add more hierarchies to folder structure. I will leave that to the testing team.
- [x] Merge `main` into `indigo-dev` and resolve potential conflicts.
- [x] Make PR, request reviews, and prepare to merge.
